### ISSUE TITLE
Heatmap scale messed up

### ISSUE THREAD

**Issue by balwierz** [mgperry/heatmaps#3]
plotHeatmaps behaves radically different depending on the range of values (all positive, all negative or a mix). This is because of discontinuity in `getScale`:
```r
getScale(-6, 0.0001)
[1] -6  6
> getScale(-6, 0)
[1] -6  0
```
A user can easily have log-transformed data which by chance can sometimes become negative (or not), leading to radically different plots depending whether there is a single datum such that `log(datum) < 0`.

**Comment by balwierz** [comment]
Also, getScale codes assumes there is something magical about zero: `getScale(-6, -5)` returns `-6, 0`

**Comment by mgperry** [comment]
Yes, I think this is how it's meant to work but I can't remember why. It should be possible to manually alter the object scale later on, this is just meant to be a convenient default (hence why zero is special).
